### PR TITLE
Update end-to-end-demo.md

### PR DESCRIPTION
**Pull request by didier-durand**
Hi there
I extended the documentation with the installation of linux-modules-extra because I had to add this package on Ubuntu 20.04 LTS (on a GCE instance): it seems that it doesn't come by default on distros / images aimed at backend servers. v4l loopback won't load if this other kernel module 'videodev' provided by the additional package is not present.

Please, test on your own to validate if needed.

Didier

### CODE REVIEW

File: docs/end-to-end-demo.md [hunk 1]
```diff
@@ -82,6 +82,10 @@ carry out one or the other, then continue on with the rest of the steps.
     curl http://deb.debian.org/debian/pool/main/v/v4l2loopback/v4l2loopback-dkms_0.12.5-1_all.deb -o v4l2loopback-dkms_0.12.5-1_all.deb
     sudo dpkg -i v4l2loopback-dkms_0.12.5-1_all.deb
     ```
+1. Install the prerequisites of v4l2loopback: it needs to be loaded simultaneously with the other kernel module 'videodev' contained in linux-modules-extra
```
Comment (kate-goldenring) [inline docs/end-to-end-demo.md:85]: Can you add this to the previous step to reduce the total number of steps?

Comment (jiria) [reply docs/end-to-end-demo.md:85]: Also, one more request. Testing this on fresh Ubuntu 20.04, package list needs to be updated and dkms installed. Can you please include these:
```
sudo apt update
sudo apt -y install linux-modules-extra-$(uname -r)
sudo apt -y install dkms
```

**Comment by didier-durand** [comment]
Hi there, I made the requested fixes. Should bbe ready to go.
Best,
Didier

File: docs/end-to-end-demo.md [hunk 2]
```diff
@@ -77,8 +77,11 @@ carry out one or the other, then continue on with the rest of the steps.
     ```
 
 ## Set up mock udev video devices
-1. Install a kernel module to make v4l2 loopback video devices. Learn more about this module [here](https://github.com/umlaeute/v4l2loopback).
+1. On a fresh Ubuntu (20.04 LTS for example), first update packages already installed. Then, install a kernel module to make v4l2 loopback video devices. It comes with prequisites: the other kernel module 'videodev' part of package linux-modules-extra and also the dkms package (Dynamic Kernel Module Support). Learn more about v4l2 loopback [here](https://github.com/umlaeute/v4l2loopback)
```
Comment (kate-goldenring) [inline docs/end-to-end-demo.md:80]: It's not required that this demo is run on Ubuntu. Only if they are using MicroK8s later. I think we can add some background about obtaining an environment before this section in a later PR. Can we simplify this to "To make dummy video4linux devices, install the v4l2loopback kernel module and its prerequisites. Learn more about ... "

Comment (didier-durand) [reply docs/end-to-end-demo.md]: @kate-goldenring : Kate. see last commit adapting text as per your suggestion

**Comment by didier-durand** [comment]
Text adapted according to @kate-goldenring PR review.  Good for merge?
Didier

**Comment by kate-goldenring** [comment]
> Text adapted according to @kate-goldenring PR review. Good for merge?
> Didier

Thank you! Looks great. Can you rebase? Then we are all set to merge.

**Review by kate-goldenring — APPROVED** [review]

**Pull request merged**